modules/app/src/main/java/org/locationtech/jtstest/function/SpatialIndexFunctions.java

```diff
@@ -117,12 +117,19 @@ public static Geometry strTreeNN(Geometry geoms, Geometry geom)
     return (Geometry) result;
   }
 
-  public static Geometry[] strTreeNNk(Geometry geoms, Geometry geom)
+  public static Geometry strTreeNNk(Geometry geoms, Geometry geom)
   {
     STRtree index = buildSTRtree(geoms);
-    int k = 10;
-    Object[] result = index.nearestNeighbour(geom.getEnvelopeInternal(), geom, new GeometryItemDistance(), k);
-    return (Geometry[]) result;
+    int k = 2;
+    Object[] knnObjects = index.nearestNeighbour(geom.getEnvelopeInternal(), geom, new GeometryItemDistance(), k);
+    Geometry[] knnGeoms = new Geometry[k];
```

Comment:
Follow the pattern here:  https://github.com/locationtech/jts/blob/master/modules/app/src/main/java/org/locationtech/jtstest/function/SpatialIndexFunctions.java#L96